src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala

```diff
@@ -82,18 +120,62 @@ class TimeSeriesRelation(config: RelationConfig)(val sqlContext: SQLContext)
     df.foreachPartition(rows => {
       implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
       val batches = rows.grouped(batchSize).toVector
-      batches.parTraverse(postRows).unsafeRunSync()
+      batches.parTraverse(updateRowsZeroSix).unsafeRunSync()
     })
 
-  private def postRows(rows: Seq[Row]): IO[Unit] = {
-    val timeSeriesItems = rows.map(r => fromRow[PostTimeSeriesItem](r))
-    post(config.apiKey, baseTimeSeriesUrl(config.project), timeSeriesItems, maxRetries)
-      .map(item => {
-        if (config.collectMetrics) {
-          timeSeriesCreated.inc(rows.length)
+  private def updateRowsZeroSix(rows: Seq[Row]): IO[Unit] = {
+    val timeSeriesItems: Seq[PostTimeSeriesItem] = rows.map(r => fromRow[PostTimeSeriesItem](r))
+    val updateTimeSeriesItems =
+      timeSeriesItems.map(i => UpdateTimeSeriesItem.fromPostTimeSeriesItem(i))
+    val updateTimeSeriesUrl =
+      uri"${baseUrl(config.project, "0.6", config.baseUrl)}/timeseries/update"
+    postOr(config.apiKey, updateTimeSeriesUrl, updateTimeSeriesItems, maxRetries) {
+      case Response(Right(body), StatusCodes.BadRequest, _, _, _) =>
+        decode[Error[TimeSeriesConflict]](body) match {
+          case Right(conflict) =>
+            resolveConflict(
+              timeSeriesItems,
+              baseTimeSeriesUrl(config.project),
+              updateTimeSeriesItems,
+              updateTimeSeriesUrl,
+              conflict.error)
+          case Left(error) => IO.raiseError(error)
         }
-        item
-      })
+    }
+  }
+
+  def resolveConflict(
+      timeSeriesItems: Seq[PostTimeSeriesItem],
+      timeSeriesUrl: Uri,
+      updateTimeSeriesItems: Seq[UpdateTimeSeriesItem],
+      updateTimeSeriesUrl: Uri,
+      duplicateIds: TimeSeriesConflict): IO[Unit] = {
+    implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+    val notFound = duplicateIds.notFound
+    val timeSeriesToUpdate = updateTimeSeriesItems.filter(p => !notFound.contains(p.id))
+    val putItems = if (timeSeriesToUpdate.isEmpty) {
+      IO.unit
+    } else {
+      put(config.apiKey, updateTimeSeriesUrl, timeSeriesToUpdate, maxRetries)
+    }
+
+    val timeSeriesToCreate = timeSeriesItems.filter(p => notFound.contains(p.id))
+    val postedItems = if (timeSeriesToCreate.isEmpty) {
```

Comment:
Just like `putItems`, this is an IO action, so it should be `postItems` (that's what it does when the action is evaluated)